### ISSUE TITLE
[bug] Doesn't generate proper bazel targets for system libraries

### ISSUE THREAD

**Issue by victorfonsec4** [conan-io/conan#12786]
### Environment details

* Operating System+version: Windows 11
* Compiler+version: bazel 6.0.0 (not a compiler but more relevant to the question at hand)
* Conan version: 1.56.0
* Python version: 3.10.5

### Steps to reproduce

1. Create an empty folder with an empty file called WORKSPACE
2. Create conanfile.py with the contents:
```
from conans import ConanFile, CMake

class ConanDeps(ConanFile):
    name = "test"
    settings = "os", "compiler", "build_type", "arch"
    generators = "BazelDeps", "BazelToolchain"
    requires = "glew/2.2.0"
```
3. run `conan install .`
4. Inspect the glew build target at `glew/BUILD`
5. try to build the target: ` bazel build //glew:glew`
6. Receive an error like:
```
Starting local Bazel server and connecting to it...
ERROR: glew/BUILD:13:11: no such package '@opengl//': The repository '@opengl' could not be resolved: Repository '@opengl' is not defined and referenced by '//glew:glew'
ERROR: Analysis of target '//glew:glew' failed; build aborted: Analysis failed
INFO: Elapsed time: 4.793s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (32 packages loaded, 123 targets configured)
    Fetching repository @local_config_cc; Restarting.
```

I was expecting that the build targets for the dependencies would be auto generate, it works for packages that have dependencies on other non system packages (i.e. packages of the form "glew/1.1.1" vs "opengl/system").

Also if you add "opengl/system" in the conanfile it doesn't generate anything.

**Comment by victorfonsec4** [comment]
In the windows case the deps @opengl and @glu should probably be replace by  something like: `linkopts = ["/DEFAULTLIB:opengl32", "/DEFAULTLIB:glu32"]` or maybe create another target that would have these linkopts?

**Comment by memsharded** [comment]
Hi @victorfonsec4 

Thanks very much for your report.

> In the windows case the deps @opengl and @glu should probably be replace by something like: linkopts = ["/DEFAULTLIB:opengl32", "/DEFAULTLIB:glu32"] or maybe create another target that would have these linkopts?

This is one of our issues with Bazel. We don't know much about it, so we rely on users advice and contributions to keep improving it. Do you know enough about Bazel? What would be your recommendation for this?

**Comment by victorfonsec4** [comment]
My knowledge is pretty limited but investigating a bit it seems possible to create a target with only linkopts, I was able to run something with:

```
cc_binary(
    name = "main",
    srcs = ["main.cc"],
    deps = [":gl_link", "glu_link"],
)

cc_library(
    name = "gl_link",
    linkopts = ["/DEFAULTLIB:opengl32"],
)

cc_library(
    name = "glu_link",
    linkopts = ["/DEFAULTLIB:glu32"],
)
```

It seems conan create a sub-directory for each non-system package with a BUILD file in it. I think it would be a matter of creating those sub-directories for the system packages also and have a BUILD file with format:
```
cc_library(
    name = "<package_name>",
    visibility = ["//visibility:public"],
    linkopts = ["<link_option>"],
)
```

 Does conan knows how to generate link options for system libraries for every compiler/os it suports?

**Comment by victorfonsec4** [comment]
It seems that there's already some code to generate the linkopts:
https://github.com/conan-io/conan/blob/f69c0269f9b7b5842f12de2ae444212b9a80a896/conan/tools/google/bazeldeps.py#L126

This should work, as long at the @targets are removed and the linkopts are added for all packages that need them. I wonder if there's a bug preventing this for executing or maybe I'm not in the right version?

**Comment by victorfonsec4** [comment]
Did a few prints, it seems `cpp_info.system_libs` is not being populated correctly, just shows as empty. At the same time the system libraries are encountered in the `self._conanfile.dependencies.host.values()[i].dependencies.direct_host.values()` list.

Might be a fault on the package recipe for not including it into system_libs?

**Comment by victorfonsec4** [comment]
So this fixes the issues but it's obviously very hacky: https://github.com/conan-io/conan/commit/bc84b0dd0e89efe8a1e141096f220fea10aa1846

Do you know if there's a better way to do this?

**Comment by Neeeflix** [comment]
Hello @memsharded, I would like to take this over.
Since you are the assignee, is this okay with you?

**Comment by memsharded** [comment]
I am fine with it, but @franramirez688 has been doing some recent efforts to improve the Bazel integration, so better wait for his feedback.

**Comment by franramirez688** [comment]
Sure, I'm refactoring the whole BazelDeps generator, so I'd like to solve this issue ASAP. Anyway, thanks a lot for the help @Neeeflix. I'll notify you when I open the PR.

**Comment by franramirez688** [comment]
This issue would be solved by this PR https://github.com/conan-io/conan/pull/14959

Adding to the `WORKSPACE` file, these lines:
```
load("@//:dependencies.bzl", "load_conan_dependencies")
load_conan_dependencies()
```
Now, all the targets are there and that error should be solved. Let me know if something else is showing up.

UPDATED:
Notice that if you use the `bazel_layout()` in your conanfile, the `WORKSPACE` would change to:
```
load("@//conan:dependencies.bzl", "load_conan_dependencies")
load_conan_dependencies()
```

**Comment by franramirez688** [comment]
Closed by https://github.com/conan-io/conan/pull/14959

It'll be released in the next Conan v1.62.
